src/main/java/net/snowflake/client/jdbc/ResultJsonParserV2.java

```diff
@@ -45,8 +45,8 @@ public void startParsing(JsonResultChunk resultChunk) throws SnowflakeSQLExcepti
     this.resultChunk = resultChunk;
     if (state != State.UNINITIALIZED)
     {
-      throw new SnowflakeSQLException(SqlState.INTERNAL_ERROR,
-                                      ErrorCode.INTERNAL_ERROR.getMessageCode(),
+      throw new SnowflakeSQLLoggedException(SqlState.INTERNAL_ERROR,
+                                      ErrorCode.INTERNAL_ERROR.getMessageCode(), null,
```

Comment:
You definitely can have a way to parse the current session to here I believe.

Reply:
added session